app/uk/gov/hmrc/voapropertylinking/repositories/DVRRepository.scala

```diff
@@ -50,17 +46,20 @@ class DVRRepository @Inject()(
   override def create(request: DetailedValuationRequest): Future[Unit] =
     Mdc.preservingMdc {
       collection
-        .insertOne(DVRRecord(request.organisationId, request.assessmentRef, request.agents))
+        .insertOne(DVRRecord(request.organisationId, request.assessmentRef, request.agents, Some(request.submissionId)))
         .toFuture()
         .map(_ => ())
         .recover {
           case e: Exception => logger.debug(e.getMessage())
         }
     }
 
-  override def exists(organisationId: Long, assessmentRef: Long): Future[Boolean] =
+  override def find(organisationId: Long, assessmentRef: Long): Future[Option[DVRRecord]] =
     Mdc.preservingMdc {
-      collection.find(or(query(organisationId))).toFuture().map(_.exists(_.assessmentRef == assessmentRef))
+      collection
```

Comment:
this is bad. I see that you didn't write this in the first place, and that it was bad when you found it, but why are we fetching ALL the dvr records and then finding the one we're interested in in-memory? Can't we let Mongo find the one we're after? We should include the assessmentRef criterium in the mongo query

Reply:
Good spot, I wasn't really examining the method that closely. I'll refactor the mongo query.